### PR TITLE
improve exception info for failed to construct managers

### PR DESCRIPTION
**Pull request by zhoney**
Change-Id: Ib6cd0d4d961a8556cee71498dbc0fcd67fc9e43c

### CODE REVIEW

File: src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java [hunk 1]
```diff
@@ -363,7 +363,11 @@ private void checkMainParams() {
                       .newInstance(info);
         } catch (Exception e) {
             throw new RuntimeException(String.format(
-                      "Construct manager failed for class '%s'", clz), e);
+                      "Construct manager failed for class '%s', please make " +
+                      "sure global arguments set correctly: " +
+                      "--url=%s,--graph=%s,--username=%s,--password=%s," +
+                      "--timeout=%s", clz, this.url(), this.graph(),
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/cmd/HugeGraphCommand.java]: clz.getSimpleName()

File: src/main/java/com/baidu/hugegraph/manager/DumpGraphManager.java [hunk 2]
```diff
@@ -59,7 +61,23 @@ public void dumpFormatter(String formatter) {
         this.dumpFormatter = Formatter.loadFormatter(formatter);
     }
 
-    public void dump(String outputDir) {
+    public void init(SubCommands.DumpGraph dump) {
+        assert dump.retry() > 0;
+        this.retry(dump.retry());
+        LocalDirectory.ensureDirectoryExist(dump.logDir());
+        this.logDir(dump.logDir());
+        this.directory = this.directory(dump.directory(), dump.hdfsConf());
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/DumpGraphManager.java]: assign in this.directory()

File: src/main/java/com/baidu/hugegraph/manager/DumpGraphManager.java [hunk 3]
```diff
@@ -59,7 +61,23 @@ public void dumpFormatter(String formatter) {
...
+
+        this.removeShardsFilesIfExists();
+        this.ensureDirectoryExist(true);
+        long splitSize = dump.splitSize();
+        E.checkArgument(splitSize >= 1024 * 1024,
+                        "Split size must >= 1M, but got %s", splitSize);
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/DumpGraphManager.java]: move to splitSize()

**Review by Linary — DISMISSED** [review]

File: src/main/java/com/baidu/hugegraph/manager/BackupRestoreBaseManager.java [hunk 4]
```diff
@@ -218,13 +218,14 @@ private InputStream inputStream(String file) {
...
-    protected Directory directory(String dir, Map<String, String> hdfsConf) {
+    protected void directory(String dir, Map<String, String> hdfsConf) {
         // Local FS directory
         if (hdfsConf == null || hdfsConf.isEmpty()) {
-            return LocalDirectory.constructDir(dir, this.graph());
+            this.directory = LocalDirectory.constructDir(dir, this.graph());
         }
         // HDFS directory
-        return HdfsDirectory.constructDir(dir, this.graph(), hdfsConf);
+        this.directory = HdfsDirectory.constructDir(dir, this.graph(),
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/manager/BackupRestoreBaseManager.java]: add else block

**Review by javeme — CHANGES_REQUESTED** [review]

**Review by javeme — APPROVED** [review]

**Review by houzhizhen — APPROVED** [review]

**Pull request merged**